esiclient/v1/cluster/cluster.py

```diff
@@ -234,64 +308,57 @@ def get_parser(self, prog_name):
-                time.sleep(15)
-                if 'tagged_network_uuids' in network_config:
-                    trunk_name = "esi-%s-trunk" % node
-                    trunk = neutron_client.find_trunk(trunk_name)
+        nodes = ironic_client.node.list(fields=['uuid', 'name', 'extra'])
+        for node in nodes:
+            extra = node.extra
+            if extra.get(ESI_CLUSTER_UUID, None) == cluster_uuid:
+                print("* Node %s" % node.name)
+                node_extra_update = []
+                node_extra_update.append(
+                    {'path': "/extra/esi_cluster_uuid",
+                     'op': 'remove'})
+
+                if ESI_PORT_UUID in extra:
+                    port_uuid = extra[ESI_PORT_UUID]
+                    print("   * deleting port %s" % port_uuid)
+                    neutron_client.delete_port(port_uuid)
+                    node_extra_update.append(
+                        {'path': "/extra/esi_port_uuid",
+                         'op': 'remove'})
+                if ESI_TRUNK_UUID in extra:
+                    trunk_uuid = extra[ESI_TRUNK_UUID]
+                    print("   * deleting trunk %s" % trunk_uuid)
+                    trunk = neutron_client.find_trunk(trunk_uuid)
                     if trunk:
-                        print("   * %s" % trunk_name)
                         utils.delete_trunk(neutron_client, trunk)
-                else:
-                    port_name = "esi-%s-%s" % (node, network.name)
-                    port = neutron_client.find_port(port_name)
-                    if port:
-                        print("   * %s" % port_name)
-                        neutron_client.delete_port(port.id)
-
-        non_uuid_node_configs = [node_config for node_config in node_configs
-                                 if 'node_uuids' not in node_config['nodes']]
-        if len(non_uuid_node_configs) > 0:
-            print("* %s node configs were skipped, as they do not"
-                  " specify specific nodes" % len(non_uuid_node_configs))
-            print("   * these nodes and ports will have to be removed"
-                  " manually")
-            print("       openstack baremetal node undeploy <node>")
-            print("       openstack port delete <port>")
+                    node_extra_update.append(
+                        {'path': "/extra/esi_trunk_uuid",
+                         'op': 'remove'})
+                if ESI_FIP_UUID in extra:
+                    fip_uuid = extra[ESI_FIP_UUID]
+                    print("   * deleting fip %s" % fip_uuid)
+                    neutron_client.delete_ip(fip_uuid)
+                    node_extra_update.append(
+                        {'path': "/extra/esi_fip_uuid",
+                         'op': 'remove'})
+                ironic_client.node.update(node.uuid, node_extra_update)
+                ironic_client.node.set_provision_state(node.uuid, 'deleted')
 
         print("UNDEPLOY COMPLETE")
```

Comment:
Can you add a message if no cluster_uuid is found, rather than showing users "UNDEPLOY COMPLETE"?

Reply:
Good idea - done!